### PR TITLE
[Quant] [PT2] Add ConvBNAdd(ReLU) Annotation into X86InductorQuantizer

### PR DESCRIPTION
**Pull request by leslie-fang-intel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #112570
* __->__ #111281
* #111280

**Summary**
This PR adds ConvBNAdd(ReLU) QAT Annotation into `X86InductorQuantizer`.

**Test Plan**
```
python -m pytest test_x86inductor_quantizer.py -k test_qat_conv2d_binary_with_quantizer_api
python -m pytest test_x86inductor_quantizer.py -k test_qat_conv2d_binary_unary_with_quantizer_api
python -m pytest test_mkldnn_pattern_matcher.py -k test_qat_qconv2d_add
python -m pytest test_mkldnn_pattern_matcher.py -k test_qat_qconv2d_add_relu
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

**Review by jgong5 — APPROVED** [review]

**Comment by leslie-fang-intel** [comment]
Hi @jerryzh168 @andrewor14, Please also kindly help to take a look of this PR.

File: test/quantization/pt2e/test_x86inductor_quantizer.py [hunk 1]
```diff
@@ -965,3 +981,86 @@ def test_qat_conv2d_unary_with_quantizer_api(self):
                     node_list,
                     is_qat=True,
                 )
+
+    @skipIfNoX86
+    def test_qat_conv2d_binary_with_quantizer_api(self):
```
Comment (jerryzh168) [inline test/quantization/pt2e/test_x86inductor_quantizer.py:986]: nit: we can remove `_with_quantizer_api` now

File: torch/ao/quantization/pt2e/qat_utils.py [hunk 2]
```diff
@@ -398,20 +420,31 @@ def _get_nodes(nodes: List[Node]) -> Tuple[Node, Node, Node, Optional[Node]]:
             if n.target == torch.ops.aten.relu.default:
                 assert relu_node is None
                 relu_node = n
+            if (n.target in [torch.ops.aten.add_.Tensor, torch.ops.aten.add.Tensor]) and (
```
Comment (jerryzh168) [inline torch/ao/quantization/pt2e/qat_utils.py:423]: OK we really should move these to quantizer..

**Review by jerryzh168 — APPROVED** [review]
LGTM. @andrewor14 please take a look as well

File: test/quantization/pt2e/test_x86inductor_quantizer.py (hunk 1 above)
Comment (leslie-fang-intel) [reply test/quantization/pt2e/test_x86inductor_quantizer.py:986]: Thanks and removed.

File: torch/ao/quantization/pt2e/qat_utils.py [hunk 3]
```diff
@@ -705,20 +738,24 @@ def _fold_conv_bn_qat_helper(m: GraphModule, is_cuda: bool) -> GraphModule:
         [True, False],  # has_relu
         [True, False],  # has_bias
         [True, False],  # relu_is_inplace
+        [True, False],  # has_add
+        [True, False],  # add_is_inplace
```
Comment (andrewor14) [inline torch/ao/quantization/pt2e/qat_utils.py:742]: Just saw this. Just wanted to point out that this exponentially increases the number of patterns we need to match against. Right now we have 6 options here + 1 for cuda vs not cuda. That's already 128 patterns. I'm adding quantized bias support here (https://github.com/pytorch/pytorch/pull/112528) which adds another dimension, bringing this up to 256 patterns. This seems concerning to me.

Comment (andrewor14) [reply torch/ao/quantization/pt2e/qat_utils.py:742]: Probably fine for these two PRs but I think we should really think about how to push some of this functionality into the subgraph rewriter so we don't have hundreds of patterns to match. cc @jerryzh168

**Pull request closed without merging**